### PR TITLE
Source google sheets: allow using spreadsheet URL instead of ID and update docs

### LINKED ISSUE #11406: Source Google Sheets: Update `spreadsheet ID` field for usability
## Tell us about the problem you're trying to solve
Today, when users implement the Google Sheets Source, most people struggle to enter the correct spreadsheet ID and have to do this several times.

## Describe the solution you’d like
For an improved UX, we should try to abstract away as much of this complexity for the user. Instead of `Spreadsheet ID`, I'd propose that we change this field to `Spreadsheet Link` so that it's easy for the user to copy paste and Airbyte can parse the ID in the backend.

## Describe the alternative you’ve considered or used
N/A

## Additional context
This is impacting the ability for users to get started in Airbyte Cloud if they have this connector.

### PR DESCRIPTION
**Pull request by grishick**
## What
 * Update wording in spec file
 * Allow using Spreadsheet URL instead of Spreadsheet ID 

## How
 * add a parser helper that extracts Sheet ID from a URL
 * add a tests for parsing sheet ID out of a URL
 * add a test for backward compatibility (parser should recognize an ID)
 * change spreadsheet_id label in spec to Spreadsheet Link per github issue 11406
 * update description text for spreadsheet_id in spec
 
## 🚨 User Impact 🚨
Before this change, users had to extract spreadsheet ID from spreadsheet URL, which was causing confusion and errors. After this change, users can use the entire spreadsheet URL and copy-paste it from their Google Sheets UI directly into connector settings without having to extract the ID from the URL

## Pre-merge Checklist
Expand the relevant checklist and delete the others.
### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
## Tests

<details><summary><strong>Unit</strong></summary>

* added test_get_spreadsheet_id
* added integration test with Spreadsheet URL

</details>

### CODE REVIEW

**Review by misteryeo — APPROVED** [review]
Approving this as is but imagine this will change once we implement: https://github.com/airbytehq/airbyte/issues/11406

File: airbyte-integrations/connectors/source-google-sheets/unit_tests/test_helpers.py [hunk 1]
```diff
@@ -227,6 +227,31 @@ def mock_client_call(spreadsheetId, includeGridData, ranges=None):
 
         self.assertEqual(expected, actual)
 
+    def test_get_spreadsheet_id(self):
+        test_url = "https://docs.google.com/spreadsheets/d/18vWlVH8BfjGegwY_GdV1B_cPP9re66xI8uJK25dtY9Q/edit#gid=1820065035"
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-google-sheets/unit_tests/test_helpers.py:231]: This should be parametrized using pytest's parametrization, but it's not a particularly high leverage use of time for you right now. 
```suggestion
        # TODO: make this parametrized
        test_url = "https://docs.google.com/spreadsheets/d/18vWlVH8BfjGegwY_GdV1B_cPP9re66xI8uJK25dtY9Q/edit#gid=1820065035"
```

**Review by sherifnada — APPROVED** [review]

**Pull request merged**